sawmill-core/src/main/java/io/logz/sawmill/processors/UrlDecodeProcessor.java

```diff
@@ -0,0 +1,108 @@
+package io.logz.sawmill.processors;
+
+import io.logz.sawmill.Doc;
+import io.logz.sawmill.ProcessResult;
+import io.logz.sawmill.Processor;
+import io.logz.sawmill.annotations.ProcessorProvider;
+import io.logz.sawmill.exceptions.ProcessorConfigurationException;
+import io.logz.sawmill.exceptions.ProcessorExecutionException;
+import io.logz.sawmill.utilities.JsonUtils;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.nio.charset.Charset;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+@ProcessorProvider(type = "urlDecode", factory = UrlDecodeProcessor.Factory.class)
+public class UrlDecodeProcessor implements Processor {
+
+    private Boolean allFields;
+    private String charSet;
+    private String field;
+    private List<String> tagOnFailure;
+
+    public UrlDecodeProcessor(Boolean allFields, String charSet, String field, List<String> tagOnFailure) {
+        this.allFields = allFields;
+        this.charSet = charSet;
+        this.field = field;
+        this.tagOnFailure = tagOnFailure;
+    }
+
+    @Override
+    public ProcessResult process(Doc doc) throws InterruptedException {
+        Map<String,Object> map = doc.getSource();
+        try{
+            if(allFields) {
+                urlDecodeMap(map);
+            }else if(doc.hasField(field)){
```

Comment:
if doc doesnt have field maybe we do want to mark it as failed?

Reply:
good idea